### PR TITLE
 use sizeof(type) instead of magic numbers

### PR DESCRIPTION
**Pull request by sebpop**

### CODE REVIEW

**Comment by gdevenyi** [comment]
Accidentally includes #311 and #317

File: arch/arm/crc32_acle.c [hunk 1]
```diff
@@ -5,39 +5,88 @@
...
         c = __crc32b(c, *buf++);
         len--;
     }
 
-    if ((len > 2) && ((ptrdiff_t)buf & 2)) {
-        buf2 = (const uint16_t *) buf;
-        c = __crc32h(c, *buf2++);
+    if ((len > sz2) && ((ptrdiff_t)buf & sz2)) {
+        uint16_t val;
+        MEMCPY(&val, buf, sizeof(val));
```
Comment (mtl1979) [inline arch/arm/crc32_acle.c]: I still don't see any valid reason to use MEMCPY() as buf is guaranteed to be always aligned.

Comment (nmoinvaz) [reply arch/arm/crc32_acle.c]: Why are you using sz2 and sz4 variables? Is there some performance improvement over just using sizeof() in all those places?

Comment (sebpop) [reply arch/arm/crc32_acle.c]: I fixed both issues.  Thanks.

**Comment by Dead2** [comment]
The sizeof commit looks good. Will pull when the dependencies are in.

**Comment by Dead2** [comment]
needs a rebase

**Comment by Dead2** [comment]
Please rebase

**Pull request merged**